btm/src/test/java/bitronix/tm/mock/NewJdbcProperUsageMockTest.java

```diff
@@ -1156,6 +1245,29 @@ public void testAutoCommitTrueWhenEnlistedButSuspended() throws Exception {
         tm.shutdown();
     }
 
+    public void testAutoCommitTrueWhenEnlistedButQuickSuspended() throws Exception {
+       Thread.currentThread().setName("testAutoCommitTrueWhenEnlistedButQuickSuspended");
+       TransactionManagerServices.getConfiguration().setQuickSuspend(true);
+       BitronixTransactionManager tm = TransactionManagerServices.getTransactionManager();
+
+       tm.begin();
+
+       Connection c = poolingDataSource1.getConnection();
+       c.prepareStatement("");
+
+       Transaction tx = tm.suspend();
+       assertNull(tm.getTransaction());
+
+       assertTrue(c.getAutoCommit());
```

Comment:
This doesn't make sense. Since the resource did not get notified that it's been detached from the transaction (i.e.: there was no XAResource.end() call) it should not report autocommit as set to true.

I'd go even one step further: if a resource got a XAResource.start() call but BitronixTransactionManager.getCurrentTransaction() returns null, then working with statements should be forbidden or you may end up with work performed on behalf of a suspended transaction.